protob/messages-monero.proto

```diff
@@ -0,0 +1,438 @@
+    optional bytes tx_enc_keys = 4;
+}
+
+/**
+ * Request: Wrapping request for transaction signature protocol.
+ * @next MoneroTsxInitResp
+ * @next MoneroTsxSetInputResp
+ * @next MoneroTsxInputsPermutationResp
+ * @next MoneroTsxInputViniResp
+ * @next MoneroTsxSetOutputResp
+ * @next MoneroTsxAllOutSetResp
+ * @next MoneroTsxMlsagDoneResp
+ * @next MoneroTsxSignInputResp
+ * @next MoneroTsxFinalResp
+ * @next MoneroRespError
+ */
+message MoneroTsxSign {
+    optional MoneroTsxInit init = 1;
+    optional MoneroTsxSetInput set_input = 2;
+    optional MoneroTsxInputsPermutation input_permutation = 3;
+    optional MoneroTsxInputVini input_vini = 4;
+    optional MoneroTsxSetOutput set_output = 5;
+    optional MoneroTsxAllOutSet all_out_set = 6;
+    optional MoneroTsxMlsagDone mlsag_done = 7;
+    optional MoneroTsxSignInput sign_input = 8;
+    optional MoneroTsxFinal final_msg = 9;
+}
+
+/**
+ * Request: Sub request of MoneroKeyImageSync. Initializing key image sync.
+ * @next MoneroKeyImageExportInitResp
+ * @next MoneroRespError
+ */
+message MoneroKeyImageExportInit {
+    optional uint64 num = 1;
+    optional bytes hash = 2;
+    repeated uint32 address_n = 3;               // BIP-32 path to derive the key from master node
+    optional uint32 network_type = 4;		     // Main-net / testnet / stagenet
+    repeated MoneroSubAddrIndicesList subs = 5;
+}
+
+/**
+ * Response: Response to key image sync initialization.
+ * @prev MoneroKeyImageExportInit
+ */
+message MoneroKeyImageExportInitResp {
+
+}
+
+/**
+ * Request: Sub request of MoneroKeyImageSync. Contains batch of the UTXO to export key image for.
+ * @next MoneroKeyImageSyncStepResp
+ * @next MoneroRespError
+ */
+message MoneroKeyImageSyncStep {
+    repeated MoneroTransferDetails tdis = 1;
+}
+
+/**
+ * Response: Response to key image sync step. Contains encrypted exported key image.
+ * @prev MoneroKeyImageSyncStep
+ */
+message MoneroKeyImageSyncStepResp {
+    repeated MoneroExportedKeyImage kis = 1;
+}
+
+/**
+ * Request: Sub request of MoneroKeyImageSync. Final message of the sync protocol.
+ * @next MoneroKeyImageSyncFinalResp
+ * @next MoneroRespError
+ */
+message MoneroKeyImageSyncFinal {
+
+}
+
+/**
+ * Response: Response to key image sync step. Contains encryption keys for exported key images.
+ * @prev MoneroKeyImageSyncFinal
+ */
+message MoneroKeyImageSyncFinalResp {
+    optional bytes enc_key = 1;
+}
+
+/**
+ * Request: Wrapping request for key image sync protocol.
+ * @next MoneroKeyImageExportInitResp
+ * @next MoneroKeyImageSyncStepResp
+ * @next MoneroKeyImageSyncFinalResp
+ * @next MoneroRespError
+ */
+message MoneroKeyImageSync {
+    optional MoneroKeyImageExportInit init = 1;
+    optional MoneroKeyImageSyncStep step = 2;
+    optional MoneroKeyImageSyncFinal final_msg = 3;
+}
+
+/**
+ * Request: Universal Monero protocol implementation diagnosis request.
+ * @next MoneroDiagResp
+ * @next MoneroRespError
+ */
+message MoneroDiag {
```

Comment:
Do we need this in production code? If not, I would rename these messages to contain `Debug` prefix to make it obvious they don't belong to production.

Reply:
I needed to discuss this. It is not needed for the production code. 
So prefixing with Debug and leaving it in the `messages-monero.proto` is OK?

Reply:
The question is whether we still need these messages or if they were only useful for development and are no longer necessary once the code has been written. If they are still needed, let's use the Debug prefix and keep them in `messages-monero` proto file.